pytorch_lightning/trainer/connectors/logger_connector/result.py

```diff
@@ -225,20 +227,23 @@ def update(self, value: _IN_METRIC, batch_size: int) -> None:
 
             if self.meta.on_step:
                 self._forward_cache = self.meta.sync(value.clone())  # `clone` because `sync` is in-place
-
-            # performance: no need to accumulate on values only logged on_step
-            if not self.meta.on_epoch:
-                self.value = self._forward_cache
-                return
+                # performance: no need to accumulate on values only logged on_step
+                if not self.meta.on_epoch:
+                    self.value = self._forward_cache
+                    return
 
             # perform accumulation with reduction
             if self.meta.is_mean_reduction:
                 self.value += value.mean() * batch_size
-                self.cumulated_batch_size += batch_size
+                # `Metric.add_state` does not work well with mypy, mypy doesn't know this is a `Tensor`
+                # we could add an assertion, but this is a hot code path
+                self.cumulated_batch_size += batch_size  # type: ignore[operator]
             elif self.meta.is_max_reduction or self.meta.is_min_reduction:
                 self.value = self.meta.reduce_fx(self.value, value.mean())
             elif self.meta.is_sum_reduction:
                 self.value += value.mean()
+            else:
+                raise ValueError(f"Invalid `reduce_fx`: {self.meta.reduce_fx!r}")
```

Comment:
I think we already have a MisconfigException when we initialize meta.

Reply:
but for mypy there needs to be an `else` I think